### PR TITLE
auto discover version

### PR DESCRIPTION
**Pull request by yogo1212**
cont https://github.com/jlund/ansible-go/pull/28

### CODE REVIEW

**Comment by yogo1212** [comment]
there's more

**Comment by yogo1212** [comment]
something about those comments in the defaults yml just makes adding hunks impossible..

**Comment by yogo1212** [comment]
k - i'm done :-p

you might want to ask about splitting this PR ;-)

**Comment by yogo1212** [comment]
is it that bad? :confused:

**Comment by yogo1212** [comment]
@jlund are you still active in this repo?

**Comment by evrardjp** [comment]
good question @yogo1212 but I haven't seen an answer to my first PR asking for the same thing. What have you done since then?

**Comment by evrardjp** [comment]
This series of patch can probably be reduced to 2 or 3 commits, but I generally like them. 👍 .

**Comment by yogo1212** [comment]
@evrardjp i've implemented the signature check, "streamlined" vars and defaults, and slightly refactored some of the code.

File: tasks/main.yml [hunk 1]
```diff
@@ -1,28 +1,26 @@
 ---
-- name: Download the Go tarball
-  get_url:
-    url: "{{ go_download_location }}"
-    dest: /usr/local/src/{{ go_tarball }}
-    checksum: "{{ go_tarball_checksum }}"
+- block:
+  - name: Discover the latest version
+    shell:
```
Comment (evrardjp) [inline tasks/main.yml:4]: probably need a > there?

File: tasks/install_go.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  set_fact:
+    go_tarball_url: "{{ go_mirror }}{{ go_tarball_name }}"
+
+- name: Fetching sha256 checksum
+  set_fact:
+    go_tarball_checksum: "sha256:{{ lookup('url',go_tarball_url+'.sha256') }}"
+  when: go_tarball_checksum is not defined
+
+- name: Create download dir
+  file: state=directory path="{{ go_download_dir }}" mode=0755
```
Comment (evrardjp) [inline tasks/install_go.yml]: it would be clearer to use the yaml format here too.

**Review by evrardjp — CHANGES_REQUESTED** [review]
Can you squash all those commits into one?

File: tasks/main.yml (hunk 1 above)
Comment (yogo1212) [reply tasks/main.yml:4]: maybe.. where exactly? :-p

File: tasks/install_go.yml (hunk 2 above)
Comment (yogo1212) [reply tasks/install_go.yml]: done that (5c02ea3)

**Comment by yogo1212** [comment]
i can squash but i'd rather wait until right before the merge.
it was surprising to see any activity at all from this repo ;-)

**Comment by evrardjp** [comment]
@yogo1212 same. But I think it's better if we are working together than having a bazillion roles doing the same thing. :)

File: tasks/main.yml (hunk 1 above)
Comment (evrardjp) [reply tasks/main.yml:4]: the shell command is on a new line, so it makes sense to me to have >- or | at line 4. But I might be wrong and ansible might be smart there. Just an habit I'd say :)

**Comment by evrardjp** [comment]
thanks for the change :)

**Comment by yogo1212** [comment]
i invite you to leave your opinion on how ansible-go should be developed in the future:
https://github.com/jlund/ansible-go/issues/37

**Pull request closed without merging**